webapp/src/ts/services/header-tabs.service.ts

```diff
@@ -69,11 +73,40 @@ export class HeaderTabsService {
       if (settings.header_tabs[tab.name].icon && settings.header_tabs[tab.name].icon.startsWith('fa-')) {
         tab.icon = settings.header_tabs[tab.name].icon;
       }
+
       if (settings.header_tabs[tab.name].resource_icon) {
         tab.resourceIcon = settings.header_tabs[tab.name].resource_icon;
       }
     });
 
     return this.tabs;
   }
+
+  async getAuthorizedTabs(settings:any = {}): Promise<HeaderTab[]> {
+    const tabs = this.get(settings);
+    const results = await Promise.all(tabs.map(tab => this.authService.has(tab.permissions)));
+
+    return tabs.filter((tab, index) => results[index]);
+  }
+
+  async getFirstAuthorizedTab(settings:any = {}): Promise<HeaderTab> {
+    const tabs = await this.getAuthorizedTabs(settings);
+
+    if (!tabs.length) {
```

Comment:
Since we're already returning `undefined` here, can't we just do: 
```suggestion
return tabs?.[0];
```

Reply:
Nice, done